torch/_refs/__init__.py

```diff
@@ -4647,6 +4648,48 @@ def triu_indices(
     )
 
 
+@register_decomposition(torch.ops.aten.bucketize)
+@out_wrapper()
+def bucketize(
+    a: TensorLikeType,
+    boundaries: TensorLikeType,
+    *,
+    out_int32: bool = False,
+    right: bool = False,
+):
+
+    out_dtype = torch.int32 if out_int32 else torch.int64
```

Comment:
Let's add a comment about this behavior. What happens if an `out` tensor is specified that's not `int32` but `out_int32` is `True`?

Reply:
In the aten implementation you get an error. I added `exact_dtype=True` to `out_wrapper` in the decomp to get same behavior.